src/hu_stuff.c

```diff
@@ -1061,10 +1064,25 @@ static void HU_UpdateCrosshair(void)
     
     crosshair_target = linetarget;
 
-    if (hud_crosshair_target && crosshair_target
-        && !(crosshair_target->flags & MF_SHADOW))
+    if (crosshair_target && !(crosshair_target->flags & MF_SHADOW))
     {
-      crosshair.cr = colrngs[hud_crosshair_target_color];
+      if (hud_crosshair_health == crosshealth_target)
+      {
+        health = crosshair_target->health;
+        
+        if (health < crosshair_target->info->spawnhealth*0.25)
+          crosshair.cr = colrngs[CR_RED];
+        else if (health < crosshair_target->info->spawnhealth*0.5)
+          crosshair.cr = colrngs[CR_BRICK];
+        else if (health < crosshair_target->info->spawnhealth*0.75)
+          crosshair.cr = colrngs[CR_GOLD];
+        else
+          crosshair.cr = colrngs[CR_GREEN];
```

Comment:
This, indeed, is different from how the health and armor numbers and the player-health-colored crosshair are colored. I like the extra bit of accuracy it adds. Should I ditch this and use the same calculations as those other cases? If not, should I use other colors?

Reply:
Please don't do that and introduce a new color-by-health scheme out of the blue. Quite the contrary, I'd like to ask you to factor out the color-by-health code into a helper function and apply it to both player and target, normalized to `spawnhealth` (if `!mobj->player`) and using the same `health_red`, `health_yellow` and `health_green` thresholds as set in the UI settings - for both player and target. Also, please don't use floating point math here.

Reply:
I assume such helper function could be used for the rest of instances where we color by health, but that'd somewhat fall outside of the scope of this PR (in fact, wouldn't it go as far as `st_stuff.c`?). Let me know what you want me to do regarding that.

Reply:
Let's just focus on coloring the crosshair patch within the scope of this PR.